pygpu/gpuarray.pyx

```diff
@@ -1430,18 +1430,21 @@ cdef GpuArray pygpu_reshape(GpuArray a, unsigned int nd, const size_t *newdims,
     if cdims == NULL:
         raise MemoryError, "could not allocate cdims"
 
-    cdef size_t d
-    for i in range(nd):
-        d = newdims[i]
-        if i == caxis:
-            d = a.size // tot
-
-            if d * tot != a.size:
-                raise GpuArrayException, "..."
-        cdims[i] = d
+    try:
+        cdef size_t d
```

Comment:
This needs to stay outside of the try..finally

Reply:
Fixed.